x/emissions/keeper/msgserver/msg_server_stake.go

```diff
@@ -281,3 +281,38 @@ func (ms msgServer) ConfirmRemoveDelegateStake(ctx context.Context, msg *types.M
 
 	return &types.MsgConfirmRemoveDelegateStakeResponse{}, nil
 }
+
+func (ms msgServer) RewardDelegateStake(ctx context.Context, msg *types.MsgRewardDelegateStake) (*types.MsgRewardDelegateStakeResponse, error) {
+	senderAddr, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return nil, err
+	}
+	// Check the target reputer exists and is registered
+	reputer, err := sdk.AccAddressFromBech32(msg.Reputer)
+	if err != nil {
+		return nil, err
+	}
+	isRegistered, err := ms.k.IsReputerRegisteredInTopic(ctx, msg.TopicId, reputer)
+	if err != nil {
+		return nil, err
+	}
+	if !isRegistered {
+		return nil, types.ErrAddressIsNotRegisteredInThisTopic
+	}
+
+	delegateInfo, err := ms.k.GetDelegateStakePlacement(ctx, msg.TopicId, senderAddr, reputer)
+	if err != nil {
+		return nil, err
+	}
+	share, err := ms.k.GetDelegateRewardPerShare(ctx, msg.TopicId, reputer)
+	if err != nil {
+		return nil, err
+	}
+	pendingReward := delegateInfo.Amount.Mul(share).Sub(delegateInfo.RewardDebt)
+	if pendingReward.GT(cosmosMath.NewUint(0)) {
+		amountInt := cosmosMath.NewIntFromBigInt(pendingReward.BigInt())
+		coins := sdk.NewCoins(sdk.NewCoin(params.DefaultBondDenom, amountInt))
+		ms.k.SendCoinsFromModuleToAccount(ctx, types.AlloraPendingRewardForDelegatorAccountName, senderAddr, coins)
```

Comment:
are you not missing an update to delegateInfo at this point? to drain their holdings and debt?

Reply:
Yes, we need to update reward debt at here!